### DISCUSSION TITLE
Showing focus session timer in the clock area

### DISCUSSION THREAD

**Discussion by tgmayfield** [ramensoftware/windhawk-mods#240] [General]
This is inspired by a misunderstanding. I tried using the System > Focus activity in Windows 11 last week, and it has a checkmark for "Show the timer in the Clock app". I thought that meant to show the timer in the taskbar clock area (on very scant evidence, definitely my fault I was wrong), but instead it fired up the clock and turned out to be annoying.

BUT, it would be so much less annoying to have my timer in the taskbar instead of the current date/time. So, my question is, how feasible is that? I know you have the mod for showing time in the clock area, but I haven't taken the time yet to review that code and see if I could adapt its ideas to this. Is that something do-able? Do you have any pointers for starting points in making that mod?

**Comment by m417z** [comment]
@tgmayfield moving it to the windhawk-mods repo, as it's related to mods and not generally to Windhawk.

> how feasible is that?

That's feasible, you can quite easily print any text on the taskbar clock by modifying the code. You can see how it is currently implemented, and how supported formatting variables are handled:
https://github.com/ramensoftware/windhawk-mods/blob/da7734260f30db5e6b11898551f6a027f0df4b03/mods/taskbar-clock-customization.wh.cpp#L731

> Do you have any pointers for starting points in making that mod?

You can start by forking it in Windhawk and playing around with it. Feel free to ask any questions here or in the [Windhawk Discord channel](https://discord.gg/WZgXScMud7).
